Ap/python log improv

a-petrini opened the pull request:
This pull request improves the readability of logs generated by Python tests by:
- changing the order of parameters passed to assert_equal() to follow a more natural ordering
- adding support for colored log messages printed on terminal

a-petrini committed:
change parameters order in assert_equal()

Parameters are now passed in a more natural order, so that the first
one is the actual value to be tested for equality, and the second one is
the expected value.

a-petrini committed:
Add colorize function for improving readability of printed logs

This function adds escape sequences to a given string, so that it is visalized in colour
when printed in an ANSI compatible terminal. Escape sequences are removed before beins saved to
debug.log files, therefore remaining unaffected by this modification

a-petrini committed:
Simplify the usage of colored mark_logs function

Now mark_logs accepts an optional parameter specifying the color that should
be used when log messages are printed on terminal. Input string is unchanged.
Messages saved on file are still unaffected.

ptagl commented on qa/rpc-tests/test_framework/util.py:
```diff
@@ -443,7 +443,7 @@ def random_transaction(nodes, amount, min_fee, fee_increment, fee_variants):
     return (txid, signresult["hex"], fee)
 
-def assert_equal(expected, actual, message = ""):
+def assert_equal(actual, expected, message = ""):
```
As a personal preference, I like the change that switched `actual` and `expected` parameters. I guess we also need at this point to do the same switch for the log string, so that it gets printed like:

`current value <actual>, expected <expected>`

As an alternative, we might also consider looking at the changes made by [ZCash](https://github.com/zcash/zcash/blob/f84a57e71bded0774bcb63e8c205c6db6ce442e1/qa/rpc-tests/test_framework/util.py#L612) and [Bitcoin](https://github.com/bitcoin/bitcoin/blob/d36bec9b3bf5c8730f14485944a035378451ae01/test/functional/test_framework/util.py#L54) to the same function.

To keep uniformity with the other util functions, I would vote to partially integrate the change by Bitcoin, in particular:
- renaming the two input parameters as `thing1` and `thing2`
- changing the debug string as `failed <thing1> == <thing2>`

drgora replied:
I definitely do not have a strong opinion on this, but I personally like the log message to point out explicitly which value is expected and which one is observed, so I'd say my weak preference goes to the first option.

a-petrini replied:
Commit e8ccdb14cc643288b109ad004751f700ada9eef8 modifies the returned string of `assert_equal(actual, expected)` function, so that values are reported in the same order of the input parameters:

`current value [left]: <actual>, expected [right]: <expected>`

Also the same commit modifies the `mark_logs()` function for supporting strings that already underwent the `colorize()` function, without propagating the ANSI escape sequences to the `debug.log` files. This is necessary for supporting fancy multicolored strings, i.e.:

`mark_logs("This is a red colored string embellished with some " + colorize('b', "blue words."), nodes, debug = 0, 'r')`

drgora reviewed (approved):
This looks good and useful to me, thanks!

a-petrini committed:
Improved clarity of the assert_equal returned message, minor mod to mark_logs

Following the suggstions and the brief discussion on the PR, the assert_equal returned
message string has been modified to reflect the order of the input parameters.
Also, mark_logs supports strings having multiple colors per line, without
propagating ANSI sequences to the debug.log file.
